dotnet/src/SemanticKernel.UnitTests/SkillDefinition/SkillCollectionTests.cs

```diff
@@ -43,4 +44,23 @@ public void ItAllowsToReplaceFunctions()
         Assert.True(target.TryGetFunction("sName", "fName", out func));
         Assert.Equal("TWO", func.Description);
     }
+
+    [Fact]
+    public void ItShouldDisposeSkillsFunctions()
+    {
+        // Arrange
+        var disposableFunction = new Mock<ISKFunction>();
+        disposableFunction.SetupGet(f => f.Name).Returns("fake-function-name");
+        disposableFunction.SetupGet(f => f.SkillName).Returns("fake-skill-name");
+        disposableFunction.As<IDisposable>();
+
+        var skillsCollection = new SkillCollection();
+        skillsCollection.AddFunction(disposableFunction.Object);
```

Comment:
What happens if you add it to a different skillsCollection, too?

Reply:
Good question Lee. The function will be registered in two collections, and it will become invalid as soon as one of the collections is disposed.

The skills collections are made disposable because, as far as I understand, they can contain functions that use system (managed or unmanaged) resources (such as HttpClient) that need to be released when the functions are no longer used.

I don't think that skills collections should be disposable because they may share the same functions, and disposing one collection will invalidate those functions. It also feels wrong to have a collection be disposable - it's like making a List<HttpClient> disposable just because it contains instances of HttpClient.

The functions should not be disposable either, because two or more functions can use the same resource, and if one function is disposed, then the resource is invalidated and can no longer be used by the other function.

I believe the correct solution to this problem is to delegate the creation of system resources to a hosting application and make it responsible for disposing them since it created and owns them. 

@shawncal @stephentoub any thoughts?

Reply:
> I believe the correct solution to this problem is to delegate the creation of system resources to a hosting application and make it responsible for disposing them since it created and owns them.

Does SK make it possible for a hosting application to know when all resources are completed?

Also how would a hosting application get at all of the resources?